src/main/java/graphql/schema/GraphQLTypeUtil.java

```diff
@@ -196,12 +196,7 @@ public static <T extends GraphQLType> T unwrapOneAs(GraphQLType type) {
      * @return the underlying type
      */
     public static GraphQLUnmodifiedType unwrapAll(GraphQLType type) {
-        while (true) {
-            if (isNotWrapped(type)) {
-                return (GraphQLUnmodifiedType) type;
-            }
-            type = unwrapOne(type);
-        }
+        return unwrapAllAs(type);
```

Comment:
This was probably a mistake to have the well  named unwrapAll return `GraphQLUnmodifiedType` which excludes GraphqlTypeReferences - but API is API so leave it as is

I did change how it works internally here